x-pack/plugins/spaces/server/spaces_service/spaces_service.ts

```diff
@@ -68,41 +74,22 @@ export class SpacesService {
       return spaceId;
     };
 
-    const internalRepositoryPromise = getStartServices().then(([coreStart]) =>
-      coreStart.savedObjects.createInternalRepository(['space'])
-    );
-
-    const getScopedClient = async (request: KibanaRequest) => {
-      const [coreStart] = await getStartServices();
-      const internalRepository = await internalRepositoryPromise;
-
-      return config$
-        .pipe(
-          take(1),
-          map((config) => {
-            const callWithRequestRepository = coreStart.savedObjects.createScopedRepository(
-              request,
-              ['space']
-            );
-
-            return new SpacesClient(
-              auditLogger,
-              (message: string) => {
-                this.log.debug(message);
-              },
-              authorization,
-              callWithRequestRepository,
-              config,
-              internalRepository,
-              request
-            );
-          })
-        )
-        .toPromise();
+    const getScopedClient = (request: RequestFacade) => {
+      if (!this.spacesClientServiceStart) {
+        throw new Error('Spaces Service has not been started yet!');
+      }
+      return this.spacesClientServiceStart?.createSpacesClient(
+        request instanceof KibanaRequest ? request : KibanaRequest.from(request)
+      );
     };
 
     return {
+      scopedClient: getScopedClient,
```

Comment:
Great question. I was going to refactor this service to stop exposing so much out of `setup`, but that ended up causing a lot of downstream changes that I wasn't comfortable keeping in this PR, so I opted against it.

> If we still want to expose this for the aforementioned reasons, can we either unify method names between setup and start or somehow rename method we expose in setup so that it's clear that no one should use it if possible?

++

Reply:
> I was going to refactor this service to stop exposing so much out of setup, but that ended up causing a lot of downstream changes that I wasn't comfortable keeping in this PR, so I opted against it.

Welp, I went and removed a bunch of functions from `setup`. It added a bit of scope to this PR, but I'm _generally_ happier with it. If you're not comfortable with the scope creep, let me know and I'll split it out into a different PR.

Reply:
Nah, let's keep it here, I believe this PR is an ideal place for a refactoring like this.